pygame_gui/elements/ui_auto_resizing_container.py

```diff
@@ -0,0 +1,258 @@
+        self.resize_top = resize_top
+        self.resize_bottom = resize_bottom
+
+        # Elements which affect the corresponding edge of the container
+        self.left_element: Optional[UIElement] = None
+        self.right_element: Optional[UIElement] = None
+        self.top_element: Optional[UIElement] = None
+        self.bottom_element: Optional[UIElement] = None
+
+        # Right to left is not the inverse of left to right because of varying sizes of elements
+        self.left_to_right_elements: List[IUIElementInterface] = []
+        self.right_to_left_elements: List[IUIElementInterface] = []
+        self.top_to_bottom_elements: List[IUIElementInterface] = []
+        self.bottom_to_top_elements: List[IUIElementInterface] = []
+
+        # An element which is anchored to the left shouldn't expand the container to the left
+        self.left_anchor_elements: List[IUIElementInterface] = []
+        self.right_anchor_elements: List[IUIElementInterface] = []
+        self.top_anchor_elements: List[IUIElementInterface] = []
+        self.bottom_anchor_elements: List[IUIElementInterface] = []
+
+    def get_left_most_point(self) -> int:
+        """
+        Gets the minimum x value any element has clamped to the min_edges_rect and max_edges_rect
+        :return: An integer
+        """
+        left = self.min_edges_rect.left
+        if self.left_element:
+            left = min(left, self.left_element.get_abs_rect().left)
+        if self.max_edges_rect is not None:
+            left = max(left, self.max_edges_rect.left)
+        return left
+
+    def get_right_most_point(self) -> int:
+        """
+        Gets the maximum x value any element has clamped to the min_edges_rect and max_edges_rect
+        :return: An integer
+        """
+        right = self.min_edges_rect.right
+        if self.right_element:
+            right = max(right, self.right_element.get_abs_rect().right)
+        if self.max_edges_rect is not None:
+            right = min(right, self.max_edges_rect.right)
+        return right
+
+    def get_top_most_point(self) -> int:
+        """
+        Gets the minimum y value any element has clamped to the min_edges_rect and max_edges_rect
+        :return: An integer
+        """
+        top = self.min_edges_rect.top
+        if self.top_element:
+            top = min(top, self.top_element.get_abs_rect().top)
+        if self.max_edges_rect is not None:
+            top = max(top, self.max_edges_rect.top)
+        return top
+
+    def get_bottom_most_point(self) -> int:
+        """
+        Gets the maximum y value any element has clamped to the min_edges_rect and max_edges_rect
+        :return: An integer
+        """
+        bottom = self.min_edges_rect.bottom
+        if self.bottom_element:
+            bottom = max(bottom, self.bottom_element.get_abs_rect().bottom)
+        if self.max_edges_rect is not None:
+            bottom = min(bottom, self.max_edges_rect.bottom)
+        return bottom
```

Comment:
Are these for internal use? I feel like for the end user they just want to be doing:

`my_container.rect.bottom` or `my_container.rect.top` and have all this already precalculated for them. Perhaps I have misunderstood the purpose.

If they are for internal use then they should also have the `_` underscore prefix.